src/main/kotlin/blackjack/domain/CardDeck.kt

```diff
@@ -0,0 +1,30 @@
+package blackjack.domain
+
+import blackjack.model.Card
+import blackjack.model.DEFAULT_CARD_DECK
+
+interface CardDeck {
+    val cards: List<Card>
+    val size: Int
+    fun shuffle()
+    fun takeOutFirstCard(): Card
+}
+
+class CardDeckImpl(
+    initialCard: List<Card> = DEFAULT_CARD_DECK,
+) : CardDeck {
+
+    private val _cards: MutableList<Card> =
+        MutableList(initialCard.size) { index -> initialCard[index].copy() }
```

Comment:
방어적 복사를 의도하신거라면 toMutableList()로도 충분하지 않을까요?
```suggestion
    private val _cards: MutableList<Card> = initialCard.toMutableList()
```

Reply:
Card를 수정하는 일이 없어서 제안해주신 것으로도 충분하긴 합니다. 다만 Card가 데이터 클래스인 점 을 감안하여서 deep copy로 미리 처리 해두는 것이 좋은 것이라고 판단하였습니다.